Haefner_IPHROS_MOGA.m

```diff
@@ -0,0 +1,508 @@
+        
+        % Solving for Q_p and other numeric values %%%%%%%%%%%%%%%%%%%%%%%
+        [Q_p_num,P_c_num,c_c_num,Q_c_num,rr_num,violation_s2] = symbolic_Q_p_stuff(c_f,T,Q_f,Am,P_f,P_p);
+        %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+        if violation_s2 == 1
+            i2 = N_e2 + 1;
+        else
+            % Storing element outputs for further calculations
+            Q_p_collection2 = [Q_p_collection2;Q_p_num]; % gpd
+            P_c_collection2 = [P_c_collection2;P_c_num]; % psi
+            c_c_collection2 = [c_c_collection2;c_c_num]; % mol/L
+            Q_c_collection2 = [Q_c_collection2;Q_c_num]; % gpd
+            rr_collection2 = [rr_collection2;rr_num];    % -
+            
+            % Increasing counter
+            i2 = i2 + 1;
+        end
+    end
+    if violation == 1 || violation_s1 == 1 || violation_s2 == 1
+        % Dependent vars
+        V_fwRO = 0; V_oRO = 0; S_oRO = 1e5; rho_oRO = 1e5; eta_ROio = 0; eta_RO = 0;
+
+        % Constraint vars
+        rr_max = 1e6; rr_min = -1e6; Q_f_max = 1e6; Q_f_min = -1e6; 
+        Q_c_max = 1e6; Q_c_min = -1e6; Q_p_max = 1e6; Q_p_min = -1e6;
+        LHS_SVM = 1e6;
+        
+    else
+        % Processing for RO System Output Values
+        V_fwRO = N_pv1*(sum(Q_p_collection)/264.172) + N_pv2*(sum(Q_p_collection2)/264.172); % m^3/day
+        
+        brine_collective = [Q_c_collection;Q_c_collection2];
+        pressure_vessel_collective = [N_pv1;N_pv2];
+        V_oRO = (brine_collective(end)/264.172)*pressure_vessel_collective(end);  % m^3/day (from gpd)
+        
+        c_c_collective = [c_c_collection;c_c_collection2];
+        pi_c = 1.12*(273+T)*2*c_c_collective(end); % psi
+        S_oRO = salinity_from_osmotic(pi_c,rho_fw,T,R,MW_salt_avg); % g/kg
+    
+        P_c_collective = [P_c_collection;P_c_collection2];
+        rho_oRO = density_from_salinity(S_oRO,P_c_collective(end),rho_fw,T,P_p); % kg/m^3
+        
+        P_f_1st_element = 1.4228424*h*(rho_sw/rho_fw);                        % psi
+        eta_ROio = 1-((P_f_1st_element-P_c_collective(end))/P_f_1st_element); % -
+    
+        rr_collective = [rr_collection;rr_collection2];
+        eta_RO_product_term = 1;
+        i_eta_RO = 1;
+        N_e = N_e1 + N_e2;
+        while i_eta_RO <= N_e
+            eta_RO_product_term = eta_RO_product_term*(1-rr_collective(i_eta_RO));
+            i_eta_RO = i_eta_RO + 1;
+        end
+        eta_RO = 1-eta_RO_product_term; % -
+    
+        %%%%%%%%%%%%%% (hyperplane constraint variable)
+        P_f_array = [P_f_collection;P_f_collection2];
+        S_f_array = [S_f_collection;S_f_collection2];
+        Q_f_array = [Q_f_collection;Q_f_collection2];
+        LHS_SVM = -1e6;
+        for i = 1:length(P_f_array)
+            var = -1334.934 + (18.222*P_f_array(i)) - (282.748*S_f_array(i)) - (200*(Q_f_array(i)/(24*264.172)));
+            if var > LHS_SVM
+                LHS_SVM = var;
```

Comment:
it seems to me this is a convoluted way of taking the maximum? If so then just take the maximum. Or better yet, since the maximum is not differentiable, add a constraint for each element individually, this will be differentiable.